flocker/provision/_install.py

```diff
@@ -44,9 +44,17 @@
                     archive_bucket=ARCHIVE_BUCKET,
                     ),
     # FLOC-1828 TODO - use ubuntu rather than ubuntu-testing
-    'ubuntu-14.04': 'https://{archive_bucket}.s3.amazonaws.com/'
-                'ubuntu-testing/14.04/$(ARCH)'.format(
-                    archive_bucket=ARCHIVE_BUCKET
+    # This could hardcode the version number instead of using ``lsb_release``
+    # but that allows instructions to be shared between versions, and for
+    # earlier error reporting if you try to install on a separate version.
+    # The $(ARCH) part must be left unevaluated, hence the backslash escapes
+    # (one to make shell ignore the $ as a substitution marker, and then
+    # doubled to make Python ignore the \ as an escape marker).
+    # The output of this value then goes into /etc/apt/sources.list which does
+    # its own substitution on $(ARCH) during a subsequent apt-get update
+    'ubuntu-14.04': 'https://{archive_bucket}.s3.amazonaws.com/ubuntu-testing/'
+                    '$(lsb_release --release --short)/\\$(ARCH)'.format(
```

Comment:
Ahah.  But of course this makes things so much more complicated.  So this is a string in which **some** pieces are meant to be interpreted by a shell and some pieces are meant to be interpreted by ... a different shell.

Um.  Yea, I dunno.  Probably the right thing to do is replace `$(lsb_release --release --short)` with `{architecture}` and do Python-side determination and substitution of that value instead of asking the shell to do it.  Then the whole value can be shell-escaped and passed to `apt-add-repository` (or whichever).  Possibly that's not worth the effort right now though.